GithubReleaseScript/github_release.py

```diff
@@ -0,0 +1,231 @@
+                         '11182210+electricRGB@users.noreply.github.com',
+                         'isabelrios@gmail.com',
+                         '41898282+github-actions[bot]@users.noreply.github.com',
+                         'litianu.razvan@gmail.com',
+                         'ralucaiordan@icloud.com',
+                         '51127880+PARAIPAN9@users.noreply.github.com']
+
+
+def fetch_raw_data(url: str) -> str:
+    return requests.get(url).content.decode("utf-8")
+
+
+def get_store_versions() -> (Version, Version):
+    """
+    Find current and previous release versions from App Store
+    Returns:
+        current_version, previous_version
+    """
+    data = fetch_raw_data(firefox_url)
+    versions = re.findall(r'"versionDisplay\\":\\"(.*?)\\",\\"releaseNotes\\', data)
+
+    return Version.build_version(versions[0]), Version.build_version(versions[1])
+
+
+def get_diff_commits(current_version: Version,
+                     previous_version: Version) -> str:
+    """
+    Get the commits difference between two versions.
+    We find the intersection with main to be able to make the diff then prettify logs
+    """
+    repo = git.Repo(search_parent_directories=True)
+    commit_origin_current = repo.commit(f"mozilla/v{current_version.string_name}")
+
+    stream = os.popen(f'git merge-base mozilla/v{previous_version.string_name} mozilla/main')
+    commit_intersect = stream.read().strip('\n')
+    # Add 1 to date to avoid getting that commit in the diff
+    commit_intersect_object = repo.commit(f"{commit_intersect}")
+    max_date = commit_intersect_object.committed_date + 1
+
+    # Pretty format doc: https://git-scm.com/docs/pretty-formats
+    return repo.git.log(f'--since={max_date}',
+                        '--pretty=format:email: %ae username: %an with commit: %s',
+                        commit_intersect,
+                        commit_origin_current)
+
+
+def filter_commits(commits: list) -> list[str]:
+    """
+    Removing commits that were made by members of the `author_exception_list`
+    """
+    contributor_commits = []
+    for commit in commits:
+        author = re.search('email: (.*) username:', commit).group(1)
+        if author not in author_exception_list:
+            contributor_commits.append(commit)
+    return contributor_commits
+
+
+def get_usernames_commits(commits: list) -> dict[str:[str]]:
+    """
+    Retrieves the @mention username as this information isn't available with local git.
+    Whenever we fetch from the GitHub API for a username, we throttle as we can hit the rate limit rapidly.
+    This is also why we temporarily save the found usernames to avoid making too much API calls.
+    Returns:
+        A dictionary containing an array of commits for specific users.
+        Key being the username, value being the array of commits (of format #12345)
+    """
+    mention_commits = {}
+    saved_users = {}
+
+    g = Github(f"{config.username}", f"{config.token}")
```

Comment:
We have talked about this offline, that would need to change so this works on a github action